// dutch-tax-box3-calculator/dist/index.js
/**
 * Dutch Tax Box 3 Calculator Library
 * @module dutch-tax-box3-calculator
 */
/**
 * Calculate Dutch Box 3 tax (wealth tax) based on savings and investments
 *
 * @param inputs - User's financial data (bank balance, investments, debts, tax partner status)
 * @param defaults - Configuration values (thresholds, tax rate, assumed return rates)
 * @returns Box 3 calculation result with taxable base, estimated tax, and breakdown
 *
 * @example
 * ```typescript
 * const result = calculateBox3Tax(
 *   {
 *     bankBalance: 50000,
 *     investmentAssets: 30000,
 *     debts: 10000,
 *     hasTaxPartner: false
 *   },
 *   {
 *     thresholds: {
 *       taxFreeAssetsPerIndividual: 57000,
 *       debtsThresholdPerIndividual: 3700
 *     },
 *     taxRate: 0.36,
 *     assumedSReturnRates: {
 *       bankBalance: 0.0092,
 *       investmentAssets: 0.0592,
 *       debts: 0.0292
 *     }
 *   }
 * );
 * ```
 */
export function calculateBox3Tax(inputs, defaults) {
    const { bankBalance = 0, investmentAssets = 0, debts = 0, hasTaxPartner = false } = inputs ?? {};
    const { thresholds, taxRate, assumedReturnRates } = defaults ?? {};
    const { taxFreeAssetsPerIndividual = 0, debtsThresholdPerIndividual = 0 } = thresholds ?? {};
    // tep 1: taxable returns calculation
    const multiplier = hasTaxPartner ? 2 : 1;
    const totalTaxFreeAllowance = multiplier * taxFreeAssetsPerIndividual;
    const totalDebtsThreshold = multiplier * debtsThresholdPerIndividual;
    const returnsOnBankBalance = bankBalance * (assumedReturnRates?.bankBalance ?? 0);
    const returnsOnInvestmentAssets = investmentAssets * (assumedReturnRates?.investmentAssets ?? 0);
    const debtAfterDeductible = Math.max(0, debts - totalDebtsThreshold);
    const totalCosts = debtAfterDeductible * (assumedReturnRates?.debts ?? 0);
    const totalReturns = returnsOnBankBalance + returnsOnInvestmentAssets;
    const taxableReturns = Math.max(0, totalReturns - totalCosts);
    // Step 2: capital yield tax base
    const capitalYieldTaxBase = bankBalance + investmentAssets - debtAfterDeductible;
    // Step 3: basis for savings & investments
    const basisForSavingsAndInvestments = Math.max(0, capitalYieldTaxBase - totalTaxFreeAllowance);
    // Step 4: share in capital yield tax base
    const shareInCapitalYieldTaxBase = capitalYieldTaxBase > 0 ? basisForSavingsAndInvestments / capitalYieldTaxBase : 0;
    // Step 5: income from savings and investments
    const incomeFromSavingsAndInvestments = taxableReturns * shareInCapitalYieldTaxBase;
    // Step 6: estimated Box 3 tax
    const estimatedBox3Tax = incomeFromSavingsAndInvestments * (taxRate ?? 0);
    return {
        taxableBase: basisForSavingsAndInvestments,
        estimatedTax: estimatedBox3Tax,
        totalReturns,
        totalCosts,
        taxableReturns,
        totalTaxFreeAllowance,
        totalDebtsThreshold,
        breakdown: [
            {
                description: 'Total bank + investment returns',
                amount: totalReturns,
            },
            {
                description: 'Total Debt costs',
                amount: totalCosts,
            },
            {
                description: 'Step 1: Total assets = bank savings + investments = ',
                amount: taxableReturns,
            },
            {
                description: 'Step 2: Capital yield tax base = total assets - deductible debts = ',
                amount: capitalYieldTaxBase,
            },
            {
                description: 'Step 3: Basis for savings & investments = capital yield tax base - allowance = ',
                amount: basisForSavingsAndInvestments,
            },
            {
                description: 'Step 4: Share in capital yield tax base = basis / capital yield tax base = ',
                amount: shareInCapitalYieldTaxBase,
            },
            {
                description: 'Step 5: Income from savings & investments = taxable returns x share = ',
                amount: incomeFromSavingsAndInvestments,
            },
        ],
    };
}
// Export constants
export { BOX3_DEFAULTS_BY_YEAR, BOX3_DEFAULTS, AVAILABLE_YEARS, DEFAULT_YEAR, getDefaultsForYear, } from './constants.js';
export default { calculateBox3Tax };
